### PR TITLE
Refresh IntelliSense time

### PR DESCRIPTION
**Pull request by andreeis**
Small change to add timing information about how long it takes for IntelliSense to refresh.

### CODE REVIEW

**Comment by pieandcakes** [comment]
@andreeis Please run the `tslinter` to fix the failed builds.

**Comment by pieandcakes** [comment]
Is this feature to to show users the time the flame is on? Should we be logging this to telemetry too(if we aren't doing it) so we can see what the trend is?

I don't know if `Flame on` should be the term for this as it has other connotations. @tara-raj Do you have a better term for this?

**Comment by andreeis** [comment]
Small change to add timing information about how long it takes for IntelliSense to refresh.

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -1017,9 +1018,12 @@ class DefaultClient implements Client {
             this.model.isTagParsing.Value = true;
             testHook.updateStatus(Status.TagParsingBegun);
         } else if (message.endsWith("Updating IntelliSense...")) {
+            timeStamp = Date.now();
             this.model.isUpdatingIntelliSense.Value = true;
             testHook.updateStatus(Status.IntelliSenseCompiling);
         } else if (message.endsWith("IntelliSense Ready")) {
+            let duration: number = Date.now() - timeStamp;
+            console.log("Flame time (sec):", duration / 1000);
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts]: This should only be logged when the loggingLevel is "Debug".
Consider also rewording this to "Refresh IntelliSense time".

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/client.ts]: I think it should be "Update IntelliSense time" to match the hover message, right?

Comment (andreeis) [reply Extension/src/LanguageServer/client.ts]: Done

**Comment by andreeis** [comment]
> Is this feature to show users the time the flame is on? Should we be logging this to telemetry too(if we aren't doing it) so we can see what the trend is?

Currently we don't collect this information as telemetry. So far, this change is useful only for visual mapping of a particular timing value with the coding operation done in VSCode. We are thinking whether is worth to collect the same info via telemetry and what is the best way to do that (aggregation criterias, more info about the operation associated with a particular timing value).

File: Extension/src/LanguageServer/client.ts [hunk 2]
```diff
@@ -1015,9 +1016,16 @@ class DefaultClient implements Client {
             this.model.isTagParsing.Value = true;
             testHook.updateStatus(Status.TagParsingBegun);
         } else if (message.endsWith("Updating IntelliSense...")) {
+            timeStamp = Date.now();
             this.model.isUpdatingIntelliSense.Value = true;
             testHook.updateStatus(Status.IntelliSenseCompiling);
         } else if (message.endsWith("IntelliSense Ready")) {
+            let settings: CppSettings = new CppSettings(this.RootUri);
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts:1023]: Did this sneak in?  I don't think you need `settings`, right?

Comment (bobbrow) [reply Extension/src/LanguageServer/client.ts:1023]: my bad, sorry it's used on the next line.

**Review by bobbrow — APPROVED** [review]

**Review by michelleangela — APPROVED** [review]

**Pull request merged**